src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -16,7 +16,7 @@ public interface MachineService {
      * @param type - any class of Machine sub class. For example: Truck.class or Bulldozer.class
      * @return the list of machines
      */
-    List<Object> getAll(Class type);
+    List<T> getAll(Class<? extends Machine> type);
```

Comment:
Consider creating a local variable in your implementation of `getAll(Class type)` method to avoid directly returning the result of `someProducer.get()`. This will make the code more readable and maintainable.